### PR TITLE
Reject reads where either the zscore or the log-likelihood is not fin…

### PR DESCRIPTION
**Pull request by nlhepler**
…ite, should address an issue report by lhon

### CODE REVIEW

File: ConsensusCore/include/ConsensusCore/Arrow/MultiReadMutationScorer.hpp [hunk 1]
```diff
@@ -252,8 +252,12 @@ namespace Arrow {
                 }
 
                 const double gs = BaselineScore();
-                const double zg = (gs - gmean) / std::sqrt(gvar);
-                const double za = (gs/nreads - gmean/nreads) / std::sqrt(gvar/nreads);
+                const double zg = (gvar == 0.0)
```
Comment (evolvedmicrobe) [inline ConsensusCore/include/ConsensusCore/Arrow/MultiReadMutationScorer.hpp]: The variance should never be zero, do you know why this is happening?

File: ConsensusCore/src/C++/Arrow/MultiReadMutationScorer.cpp [hunk 2]
```diff
@@ -310,7 +310,7 @@ namespace Arrow {
 
             const double zscore = (ll - mean) / std::sqrt(var);
 
-            if (isinf(ll) || zscore < threshold)
+            if (!std::isfinite(ll) || !std::isfinite(zscore) || zscore < threshold)
```
Comment (evolvedmicrobe) [inline ConsensusCore/src/C++/Arrow/MultiReadMutationScorer.cpp]: Same here.Any idea?

File: ConsensusCore/include/ConsensusCore/Arrow/MultiReadMutationScorer.hpp (hunk 1 above)
Comment (nlhepler) [inline ConsensusCore/include/ConsensusCore/Arrow/MultiReadMutationScorer.hpp]: Yes, we have no reads.

File: ConsensusCore/src/C++/Arrow/MultiReadMutationScorer.cpp (hunk 2 above)
Comment (nlhepler) [inline ConsensusCore/src/C++/Arrow/MultiReadMutationScorer.cpp]: I haven't looked at lhon's examples yet, but I suspect var is getting set to nan or -nan or something.

**Pull request merged**